src/serverenvironment.cpp

```diff
@@ -795,6 +795,8 @@ struct ActiveABM
 	int chance;
 	std::vector<content_t> required_neighbors;
 	bool check_required_neighbors; // false if required_neighbors is known to be empty
+	std::vector<content_t> without_neighbors;
+	bool check_without_neighbors; // false if without_neighbors is known to be empty
```

Comment:
fwiw this is pointless, checking if a vector is empty is `O(1)`

Reply:
It is reused code from `bool check_required_neighbors`.

How should it be done?

Reply:
remove `check_x_neighbors` and instead test for `!x_neighbors.empty()`
but not needed here now, can be cleaned up later.